File: 13-RomanToInteger/13-RomanToInteger.c
// Last updated: 8/20/2025, 5:44:42 PM
int romanToInt(char* s) {
    int romanValues[256] = {0}; // ASCII mapping for quick lookup
    romanValues['I'] = 1;
    romanValues['V'] = 5;
    romanValues['X'] = 10;
    romanValues['L'] = 50;
    romanValues['C'] = 100;
    romanValues['D'] = 500;
    romanValues['M'] = 1000;

    int total = 0;
    int prevValue = 0;
    
    while (*s) {
        int currentValue = romanValues[*s];
        
        if (currentValue > prevValue) {
            total += currentValue - 2 * prevValue;  // Correct previous addition
        } else {
            total += currentValue;
        }
        
        prevValue = currentValue;
        s++;
    }
    
    return total;
}

File: 13-RomanToInteger/13-RomanToInteger.py
# Last updated: 8/20/2025, 5:44:36 PM
class Solution:
    def romanToInt(self, s: str) -> int:
        roman = {
            'I': 1,
            'V': 5,
            'X': 10,
            'L': 50,
            'C': 100,
            'D': 500,
            'M': 1000
        }
        
        total = 0
        prev_value = 0
        
        for char in reversed(s):  # iterate from right to left
            value = roman[char]
            if value < prev_value:
                total -= value
            else:
                total += value
            prev_value = value
        
        return total